### PR TITLE
[sgl-kernel] chore: update deepgemm version

### PR DESCRIPTION
**Pull request by FlamingoPg**
## Motivation

As titled.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

**Comment by hlu1** [comment]
cc @YAMY1234 to verify perf improvement to dsv32

**Comment by YAMY1234** [comment]
Verified around 10% improvement for kernel deep_gemm.fp8_mqa_logits
Before:
image

After:
image

**Comment by hlu1** [comment]
@FlamingoPg  There is one more update coming: https://github.com/deepseek-ai/DeepGEMM/pull/230

**Review by Fridge003 — APPROVED** [review]
